dynatrace-aws-logs.sh

```diff
@@ -199,13 +199,15 @@ arguments:
     UseExistingActiveGate="$USE_EXISTING_ACTIVE_GATE" TenantId="$TENANT_ID" DynatracePaasToken="$TARGET_PAAS_TOKEN" \
     --no-fail-on-empty-changeset
 
+  # SHOW OUTPUTS
+  aws cloudformation describe-stacks --stack-name "$STACK_NAME" --query "Stacks[0].Outputs"
```

Comment:
Output needs to be at the end. One of output values is referenced in the docs (new AG setup, you take EC2 hostname to find the AG on AG list)

Reply:
What I am trying to accomplish here is that there is a CloudFormation phase, and here's it's output (including anything like `No changes to deploy. Stack BeluPerfLogsCLientSecond is up to date`). I don't want this message to be the last when there are no changes, as it IS confusing and I won't be persuaded otherwise.

My idea was, that I want to say clearly that something is still happening apart from the stack's deployment (which says "No changes"). So I printed something to suggest "And now, after those no changes, something more: a Lambda update". I'm not deleting the whole of "Deploying Lambda changes", unless you give me a constructive suggestion how I can achieve the same purpose in another way.

What we can discuss:
I considered value of LAMBDA_ARN no harm and a nice context bonus, since the fact we print it in the output is already pretty obsolete. I can delete the arn if it spares me the argument. 
Another thing that is also not necessary is that I also redirected Lambda update output from dev/null to stdout, as I thought the message Successful a bonus to consider, and subject to reviewers opinions.

https://www.dynatrace.com/support/help/shortlink/aws-log-fwd#deploy
I assume you mean this doc, point 3 - in that case I'm pretty sure you've mentioned today that we either way want to improve that step and make it unnecessary. In the meantime, if we revert to dev/nulling Lambda output, then your Output is still last line but one and no user should have worse trouble finding it then they already have.

Reply:
I am fine with all this! Lambda arn is nice bonus. The info that lambda is being deployed is very much needed
My view:
1. I need stack outputs at the end, I think this doc reference to one of its values stays for now
2. I think that "aws lambda update-function-code" json output is not needed

Reply:
Sorry if I did not make this clear enough in my earlier comments

Reply:
I agree with @mmajcher that these outputs should be shown as the last message of the deployment script as they may be needed for subsequent steps and we don't want users to search for them in the middle of the deployment logs

Reply:
Please check if that would be ok:

```
[cloudshell-user@ip-10-1-157-79 ~]$ ./dynatrace-aws-logs.sh deploy --use-existing-active-gate true

Deployment script will use following parameters:
TARGET_URL="https://ec2-18-232-177-51.compute-1.amazonaws.com:9999/e/jxw01498", TARGET_API_TOKEN=, USE_EXISTING_ACTIVE_GATE="true", TARGET_PAAS_TOKEN=, REQUIRE_VALID_CERTIFICATE="false", STACK_NAME="BeluPerfLogsCLientSecond"
Deploying stack BeluPerfLogsCLientSecond

Waiting for changeset to be created..

No changes to deploy. Stack BeluPerfLogsCLientSecond is up to date

Deployed Lambda changes to arn:aws:lambda:us-east-1:908047316593:function:BeluPerfLogsCLientSecond-Lambda-oxy1d5DDH2FV

[
{
"OutputKey": "FirehoseArn",
"OutputValue": "arn:aws:firehose:us-east-1:908047316593:deliverystream/BeluPerfLogsCLientSecond-FirehoseLogStreams-yFbrfo6WxsC8",
"Description": "Firehose ARN"
},
{
"OutputKey": "LambdaArn",
"OutputValue": "arn:aws:lambda:us-east-1:908047316593:function:BeluPerfLogsCLientSecond-Lambda-oxy1d5DDH2FV",
"Description": "Lambda ARN"
},
{
"OutputKey": "CloudWatchLogsRoleArn",
"OutputValue": "arn:aws:iam::908047316593:role/BeluPerfLogsCLientSecond-CloudWatchLogsRole-SDQBYLQCVDEN",
"Description": "CloudWatch Logs role ARN allowing streaming to Firehose"
}
]

```

Reply:
Hmm, now that I look at it, it's a bit contradictory that the CLI output says `No changes to deploy` and immediately after we say `Deployed Lambda changes ...`. Maybe let's change it up to `Updated lambda code of $LAMBDA_ARN`?

Reply:
Like it better?

```
[cloudshell-user@ip-10-1-157-79 ~]$ ./dynatrace-aws-logs.sh deploy --use-existing-active-gate true

Deployment script will use following parameters:
TARGET_URL="https://ec2-18-232-177-51.compute-1.amazonaws.com:9999/e/jxw01498", TARGET_API_TOKEN=, USE_EXISTING_ACTIVE_GATE="true", TARGET_PAAS_TOKEN=, REQUIRE_VALID_CERTIFICATE="false", STACK_NAME="BeluPerfLogsCLientSecond"
Deploying stack BeluPerfLogsCLientSecond

Waiting for changeset to be created..

No changes to deploy. Stack BeluPerfLogsCLientSecond is up to date

Updated Lambda code of arn:aws:lambda:us-east-1:908047316593:function:BeluPerfLogsCLientSecond-Lambda-oxy1d5DDH2FV

[
{
"OutputKey": "FirehoseArn",
"OutputValue": "arn:aws:firehose:us-east-1:908047316593:deliverystream/BeluPerfLogsCLientSecond-FirehoseLogStreams-yFbrfo6WxsC8",
"Description": "Firehose ARN"
},
{
"OutputKey": "LambdaArn",
"OutputValue": "arn:aws:lambda:us-east-1:908047316593:function:BeluPerfLogsCLientSecond-Lambda-oxy1d5DDH2FV",
"Description": "Lambda ARN"
},
{
"OutputKey": "CloudWatchLogsRoleArn",
"OutputValue": "arn:aws:iam::908047316593:role/BeluPerfLogsCLientSecond-CloudWatchLogsRole-SDQBYLQCVDEN",
"Description": "CloudWatch Logs role ARN allowing streaming to Firehose"
}
]
```

Reply:
![image](https://user-images.githubusercontent.com/50921490/125431309-62d6f772-bc1c-49ab-a408-486f18f756f2.png)

Reply:
I like the output better but I still think this message should be below the line `aws lambda update-function-code` in code :P